### DISCUSSION TITLE
Color Decorator / picker not displayed with extension active

### DISCUSSION THREAD

**Discussion by dcog989** [redhat-developer/vscode-xml#976] [Q&A] [answered]
The VS Code Color Decorator / picker is not displayed on hover in XML files when I have this extension enabled. It does display if I disable it.

I've been through the settings but can't see anything to cause the problem. Any suggestions please?

**Comment by fbricon** [comment]
Can you please share some sample xml and add a screenshot with/without vscode-xml enabled?

**Reply by dcog989** [reply]
Screenshots with XML active / inactive:

![Screenshot 2024-03-14 195713](https://github.com/redhat-developer/vscode-xml/assets/89043002/6cec66e0-f2b8-4640-9627-f26d56704b4d)
![Screenshot 2024-03-14 195743](https://github.com/redhat-developer/vscode-xml/assets/89043002/fc6faee9-59bf-48d4-8283-b8546da36043)

Sample code:

```
<Ditto_Theme_File Version = "3" Author = "Bubba" Notes = "">

	<color>#fff</color>

	<CaptionLeft>RGB(122, 150, 223)</CaptionLeft>
	<CaptionRight>RGB(157, 185, 235)</CaptionRight>

	<colortest>#fff</colortest>

</Ditto_Theme_File>
```

Note - I can't get it to work with RGB regardless.

**Reply by angelozerr** [reply]
At first,vscode-xml has a color support https://github.com/redhat-developer/vscode-xml/blob/main/docs/Features/XMLColorsFeatures.md

For your problem, is it working with CSS or HTML files?

**Reply by dcog989** [reply]
Hi. Yeah, it works perfectly with CSS, HTML, JS, JSX, YAML, etc. Only XML seems to be an issue.

I've spent hours on this but still might be missing the obvious! I think this comment might offer a clue - https://github.com/microsoft/vscode/issues/180433#issuecomment-1632290767 ???

**Comment by angelozerr** [comment]
I have installed `Color Picker Universal` and it is working for me:

![image](https://github.com/redhat-developer/vscode-xml/assets/1932211/3404e9b5-519c-4778-881a-ab5921f9dafb)

vscode-xml does nothing about color, the https://github.com/redhat-developer/vscode-xml/blob/main/docs/Features/XMLColorsFeatures.md uses the language server support from vscode which is used too in CSS files.

Which version of vscode are you using?
Which version of vscode-xml are you using?

**Reply by dcog989** [reply]
I've just tried 'Color Picker Universal' again and everything now works perfectly! I can't explain why it was not before.

However, I thought that Color (Decorator) picker should 'just work'? I shouldn't need additional extensions?

For reference - I'm using latest production versions of  vscode (1.87.2) and vscode-xml (0.26.1).

Thanks for your help with this. :)

**Reply by angelozerr** [reply, marked as answer]
If I understand your need, you would like write this XML:

```xml
<Ditto_Theme_File Version = "3" Author = "Bubba" Notes = "">
	<color>#fff</color>
</Ditto_Theme_File>
```

and having color support when you write `#fff` anywhere? If it that it doesn't work like this. For CSS it is the same thing. If you declare `#fff` not after a color attribute you should not see color.

For vscode-xml it is the same thing, you need to configure where color can be occur, please read https://github.com/redhat-developer/vscode-xml/blob/main/docs/Features/XMLColorsFeatures.md and you will not need another extension.

**Reply by dcog989** [reply]
OK. That makes sense. Many thanks for your help. :)
